src/ntfsck.c

```diff
@@ -1817,20 +1819,90 @@ static int ntfsck_truncate_attr(ntfs_attr *na, s64 new_size)
 	return 0;
 }
 
-static int ntfsck_check_directory(ntfs_inode *ni)
+static int ntfsck_check_non_resident_attr(ntfs_attr *na, struct rl_size *out_rls)
 {
+	BOOL need_fix = FALSE;
 	ntfs_volume *vol;
+	ntfs_inode *ni;
+	struct rl_size rls = {0, };
+
+	if (!na || !na->ni || !na->ni->vol)
+		return STATUS_ERROR;
+
+	ni = na->ni;
+	vol = na->ni->vol;
+
+	/* check cluster runlist and set bitmap */
+	if (ntfsck_decompose_setbit_runlist(na, &rls, &need_fix)) {
+		check_failed("Failed to get non-resident attribute(%d) "
+				"in directory(%lld)",
+				na->type, (unsigned long long)ni->mft_no);
+		return STATUS_ERROR;
+	}
+
+	/* if need_fix is set to TRUE, apply modified runlist to cluster runs */
+	if (need_fix == TRUE) {
+		check_failed("Non-resident cluster run of inode(%lld:%d) "
+				"corrupted. truncate it",
+				(unsigned long long)ni->mft_no, na->type);
+
+		if (ntfsck_ask_repair(vol)) {
+			s64 tr_size;
+
+			/*
+			 * keep a valid runlist as long as possible.
+			 * if truncate zero, call with second parameter to 0
+			 */
+			tr_size = rls.asize;
+			if (ntfsck_truncate_attr(na, tr_size))
+				return STATUS_ERROR;
+
+			fsck_err_fixed();
+		}
+	}
+
+#if 0
```

Comment:
test code ? If this code is not needed, please remove it.

Reply:
It removed in latest patchset